reyna/src/com/b2msolutions/reyna/services/StoreService.java

```diff
@@ -53,10 +65,16 @@ protected void onHandleIntent(Intent intent) {
 	}
 
 	private void insert(Message message) {
-        Logger.v(TAG, "insert");
-		
+		Logger.v(TAG, "insert");
+
 		try {
-			this.repository.insert(message);
+			if (StoreService.storageSizeLimit == null) {
+				this.repository.insert(message);
+			}
+			else {
+				this.repository.insert(message, StoreService.storageSizeLimit);
```

Comment:
It can be one call to insert, then inside insert it check for storage limit and apply replace when size < limit -threshold.

@Youhana-Hana